tests/testthat/test-ops-integer-division.R

```diff
@@ -1,69 +1,25 @@
 context("Integer division operations")
 
 test_that("integer division works for interval numerator", {
-  int1 <- ymd("2010-01-01") %--% ymd("2011-01-01")
-  expect_error(int1 %/% int2)
-  expect_equal(int1 %/% months(1), 12)
-  expect_error(months(12) %/% int1)
-  expect_equal(int1 %/% ddays(1), 365)
-  expect_error(ddays(365) %/% int1)
-})
-
-
-test_that("integer division works for interval numerator with vectors", {
-  int1 <- ymd("2010-01-01") %--% ymd("2011-01-01")
-  int2 <- ymd("2009-01-01") %--% ymd("2011-01-01")
-  int3 <- ymd("2009-01-01") %--% ymd("2010-01-01")
-  expect_error(int2 %/% c(int1, int3))
-  expect_error(c(int1, int2) %/% int3)
-  expect_equal(int1 %/% months(1:2), c(12, 6))
-  expect_error(months(c(6, 12)) %/% int1)
-  expect_equal(int1 %/% ddays(1:2), c(365, 182))
-  expect_equal(c(int1, int2) %/% ddays(2), c(182, 365))
-})
-
-
-test_that("integer division works for period numerator", {
-  int1 <- ymd("2010-01-01") %--% ymd("2010-12-31")
-  expect_error(years(2) %/% int1)
-  expect_equal(years(1) %/% days(1), 365)
-  expect_equal(months(13) %/% years(1), 1)
-  expect_error(months(1) %/% ddays(1))
-})
-
-test_that("integer division works for period numerator with vectors", {
-  int1 <- ymd("2010-01-01") %--% ymd("2011-01-01")
-  int2 <- ymd("2009-01-01") %--% ymd("2011-01-01")
-  expect_error(years(2) %/% c(int1, int2))
-  expect_error(years(1:2) %/% int1)
-  expect_equal(years(1) %/% months(1:2), c(12, 6))
-  expect_equal(months(c(6, 12)) %/% years(1), c(0, 1))
-  expect_error(years(1) %/% ddays(1:2))
-  expect_error(years(1:2) %/% ddays(2))
-})
-
-test_that("integer division works for duration numerator", {
-  int1 <- ymd("2010-01-01") %--% ymd("2011-01-01")
-  expect_error(dyears(2) %/% int1)
-  expect_error(dyears(1) %/% months(1))
-  expect_equal(dyears(1) %/% ddays(1), 365)
-})
-
-test_that("integer division works for duration numerator with vectors", {
-  int1 <- ymd("2010-01-01") %--% ymd("2011-01-01")
-  int2 <- ymd("2009-01-01") %--% ymd("2011-01-01")
-  expect_error(dyears(2) %/% c(int1, int2))
-  expect_error(dyears(1:2) %/% int1)
-  expect_error(dyears(1) %/% months(1:2))
-  expect_error(dyears(1:2) %/% years(1))
-  expect_equal(dyears(1) %/% ddays(1:2), c(365, 182))
-  expect_equal(dyears(1:2) %/% ddays(2), c(182, 365))
-})
-
 
-test_that("integer division works with Period denominator (#235)", {
-  a <- interval(ymd(20140331), ymd(20140519))
-  expect_equal(a %/% months(1), 1)
-  a <- interval(ymd(20140331), ymd(20140419))
-  expect_equal(a %/% months(1), 0)
+  verify_output(test_path("test-ops-integer-division.txt"), {
+    int <- ymd("2010-01-01") %--% ymd("2010-06-01")
```

Comment:
If I understand the change correctly, the new tests constitute a subset of the original tests. Each previous block included a test for recycling. 

Also, call me old school but I don't  quite like this monolitic approach to testing. It's harder to debug, keeps output outside of the test and it's quite more fragile. Small output changes on our or R will cause headaches.

Reply:
I think I misgeneralised the intent of the tests after reading a few; you are correct that some are also testing that the recycling rules have been correctly interpreted. I'll add those back in.

But you've misinterpreted the intent of this test; it makes sense as a single complicated because it's easier to see that all 9 cases are covered. The use of `verify_output()` turns it into a regression test, which I think makes more sense when you are testing user facing behaviour, which here is exactly which error message you get (the old use of `expect_error()` is dangerous because any error, even if unexpected, will cause the test to pause). `verify_output()` is designed for testing user output so it doesn't fail on CRAN, making it less fragile and easy to update. For example, if we merge #827 after this PR, it'll be easy to see exactly what's changed.